rpcs3/CMakeLists.txt

```diff
@@ -160,11 +160,18 @@ if(NOT LLVM_FOUND)
 else()
 	add_definitions(${LLVM_DEFINITIONS})
 	add_definitions(-DLLVM_AVAILABLE)
-	if (CMAKE_BUILD_TYPE STREQUAL "Release")
-		llvm_map_components_to_libnames(LLVM_LIBS mcjit vectorize ipo x86codegen x86disassembler)
+	
+	#Use static linking for windows binary but always dynamic on *nix
+	if (WIN32 OR MSVC)
+		if (CMAKE_BUILD_TYPE STREQUAL "Release")
+			llvm_map_components_to_libnames(LLVM_LIBS mcjit vectorize ipo x86codegen x86disassembler)
+		else()
+			llvm_map_components_to_libnames(LLVM_LIBS mcjit vectorize ipo x86codegen x86disassembler mcdisassembler)
+		endif()
 	else()
-		llvm_map_components_to_libnames(LLVM_LIBS mcjit vectorize ipo x86codegen x86disassembler mcdisassembler)
+		set(LLVM_LIBS LLVM-4.0.so)
```

Comment:
are you sure it can't be done in a less hacky way (i.e by using `LLVM_LINK_LLVM_DYLIB` or similar. Or by using some other variable provided by `find_package`.